src/engraving/libmscore/undo.h

```diff
@@ -943,6 +943,8 @@ class RemoveExcerpt : public UndoCommand
     void undo(EditData*) override;
     void redo(EditData*) override;
 
+    std::vector<const EngravingObject*> objectItems() const override;
```

Comment:
What is the purpose of this change? And why is it needed only for RemoveExcerpt and not for AddExcerpt?

Reply:
objectItems() specifies which engraving objects will be changed when using this command. Without this change, we can't send data to changesChannel() (as ScoreChangesRange will be considered invalid), so the score will not be marked as unsaved. See this code in Score::endCmd:
    
   ```
 if (!rollback && range.isValid()) {
        changesChannel().send(range);
    }
```

I added the same change for AddExcerpt, thanks

Reply:
Hm... I see. But isn't that a bit dangerous? It looks like there are more actions without object items, but simply every action should trigger the file to become unsaved.

Reply:
I moved the range.isValid() check directly to PlaybackModel. Now even if we failed to resolve ScoreChangesRange from some command, we still mark the project as unsaved

Reply:
Seems good to me!